lib/nrf_modem_lib/trace_backends/ram/ram.c

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <stdio.h>
+#include <string.h>
+#include <sys/errno.h>
+#include <zephyr/kernel.h>
+#include <zephyr/sys/util.h>
+#include <zephyr/sys/ring_buffer.h>
+#include <zephyr/logging/log.h>
+#include <modem/trace_backend.h>
+
+LOG_MODULE_REGISTER(modem_trace_backend, CONFIG_MODEM_TRACE_BACKEND_LOG_LEVEL);
+
+/* don't put this in noinit, memfault won't be able to capture it */
+#define RING_BUF_DECLARE_STATIC(name, size8) \
```

Comment:
Not sure we need to use a macro here. Might be more readable to just declare it directly